rajawali/src/main/java/c/org/rajawali3d/engine/RenderView.java

```diff
@@ -0,0 +1,66 @@
+package c.org.rajawali3d.engine;
+
+import android.graphics.Rect;
+import android.support.annotation.Nullable;
+import c.org.rajawali3d.annotations.GLThread;
+
+/**
+ * Interface to be implemented by views that are to be managed by the Engine. A RenderView represents
+ * a viewport on the render surface
+ *
+ * @author Jared Woolston (Jared.Woolston@gmail.com)
+ */
+public interface RenderView {
+
+    /**
+     * Sets the active {@link Engine} this {@link RenderView} is registered with, or clears it if
+     * {@code null} is provided. There is no guarantee of this method being called on the GL thread.
+     *
+     * @param engine The active {@link Engine} or {@code null}.
+     */
+    void setEngine(@Nullable Engine engine);
+
+    /**
+     *
+     * @return the {@link RenderModel} instance presented in this view, if any
+     */
+    @Nullable
+    RenderModel getRenderModel();
+
+    void setViewport(Rect rect);
```

Comment:
Can this ever be null? We should use the appropriate annotation here regardless of the answer (`@NonNull` or `@Nullable`)